uberfire-extensions/uberfire-metadata/uberfire-metadata-backends/uberfire-metadata-backend-infinispan/src/main/java/org/uberfire/ext/metadata/backend/infinispan/proto/KObjectMarshaller.java

```diff
@@ -0,0 +1,287 @@
+
+        String id = protoStreamReader.readString(MetaObject.META_OBJECT_ID);
+        String type = protoStreamReader.readString(MetaObject.META_OBJECT_TYPE);
+        String clusterId = protoStreamReader.readString(CLUSTER_ID);
+        String segmentId = protoStreamReader.readString(SEGMENT_ID);
+        String key = protoStreamReader.readString(MetaObject.META_OBJECT_KEY);
+        String fullText = protoStreamReader.readString(MetaObject.META_OBJECT_FULL_TEXT);
+
+        return new KObjectImpl(id,
+                               type,
+                               clusterId,
+                               segmentId,
+                               key,
+                               properties,
+                               !StringUtils.isEmpty(fullText));
+    }
+
+    @Override
+    public void writeTo(ProtoStreamWriter protoStreamWriter,
+                        KObject kObject) throws IOException {
+
+        protoStreamWriter.writeString(MetaObject.META_OBJECT_ID,
+                                      kObject.getId());
+        protoStreamWriter.writeString(MetaObject.META_OBJECT_TYPE,
+                                      kObject.getType().getName());
+        protoStreamWriter.writeString(CLUSTER_ID,
+                                      kObject.getClusterId());
+        protoStreamWriter.writeString(SEGMENT_ID,
+                                      kObject.getSegmentId());
+        protoStreamWriter.writeString(MetaObject.META_OBJECT_KEY,
+                                      kObject.getKey());
+
+        List<KProperty<?>> props = new ArrayList<>();
+        kObject.getProperties().iterator().forEachRemaining(props::add);
+
+        props.removeIf(kProperty -> props.stream()
+                .filter(prop -> kProperty.getName().equals(prop.getName())).count() > 1);
+
+        props.forEach(kProperty -> {
+            try {
+                this.build(toProtobufFormat(kProperty.getName()),
+                           kProperty.getValue(),
+                           protoStreamWriter);
+            } catch (IOException e) {
+                logger.error("error",
+                             e);
+            }
+        });
+
+        if (kObject.fullText()) {
+            protoStreamWriter.writeString(MetaObject.META_OBJECT_FULL_TEXT,
+                                          props.stream()
+                                                  .filter(kProperty -> kProperty.isSearchable() && !(kProperty.getValue() instanceof Boolean))
+                                                  .map(kProperty -> String.valueOf(kProperty.getValue()).toLowerCase())
+                                                  .collect(joining("\n")));
+        }
+    }
+
+    @Override
+    public Class<? extends KObjectImpl> getJavaClass() {
+        return KObjectImpl.class;
+    }
+
+    @Override
+    public String getTypeName() {
+        return this.typeName;
+    }
+
+    private void build(String name,
```

Comment:
This method name is misleading. You're not building anything, just writing name-value pairs to ProtoStreamWriter :)